in_toto/gpg/dsa.py

```diff
@@ -183,6 +183,8 @@ def gpg_verify_signature(signature_object, pubkey_info, content,
     hash_algorithm_id:
             one of SHA1, SHA256, SHA512 (see in_toto.gpg.constants) used to
             verify the signature
+            NOTE: Overrides any hash algorithm specification in "pubkey_info"'s
+            "hashes" or "method" fields.
```

Comment:
I wonder if this note sounds more dramatic than it really is (as far as I understand `hash_algorithm_id` is populated with `pubkey_info["hashes"]` or `"method"` fields when not used directly?

Reply:
No. Currently `SHA256` is passed explicitly in the interface function `gpg.functions.gpg_verify_signature` to the corresponding dsa or rsa verification function. While it aligns with the possible values for `"hashes"` or `"method"`, assigned in `gpg.common.parse_pubkey_payload`, it is not populated from those fields.

But even if it were populated from those fields (IMO a good idea), we still want to be able to use a public key that we restrict to sha256 for binary signatures (i.e. `key_info["hashes"] = "pgp+SHA2"`), to verify self-signatures with other hash algos ... hence the support for an override and hence the comment. :)

We can of course re-think the fields in the public key dict and how we assign them. But I wouldn't do it as part of this PR.